File: 01_gpio_digital/led_blink_task.py
import RPi.GPIO as GPIO
import time

RED = 2
YELLOW = 3
GREEN = 4
GPIO.setmode(GPIO.BCM)  # GPIO.BCM or GRIO.BOARD
GPIO.setup(RED, GPIO.OUT)   # GPIO.OUT or GPIO.IN
GPIO.setup(YELLOW, GPIO.OUT)
GPIO.setup(GREEN, GPIO.OUT)

GPIO.output(RED, GPIO.HIGH) # 1
print("led on")
time.sleep(2)
GPIO.output(RED, GPIO.LOW) # 0
print("led off")

GPIO.output(YELLOW, GPIO.HIGH) # 1
print("led on")
time.sleep(2)
GPIO.output(YELLOW, GPIO.LOW) # 0
print("led off")

GPIO.output(GREEN, GPIO.HIGH) # 1
print("led on")
time.sleep(2)
GPIO.output(GREEN, GPIO.LOW) # 0
print("led off")

GPIO.setup(RED, GPIO.IN)
GPIO.setup(YELLOW, GPIO.IN)
GPIO.setup(GREEN, GPIO.IN)

GPIO.cleanup()  # GPIO 핀상태 초기화

File: RaspberryPi/01_gpio_digital/led_blink_task.c
#include <wiringPi.h>
#define RED 2
#define YELLOW 3
#define GREEN 4
int main (void)
{
  // wiringPiSetup();
  wiringPiSetupGpio();
  pinMode (RED, OUTPUT);
  pinMode (YELLOW, OUTPUT);
  pinMode (GREEN, OUTPUT);

  digitalWrite (RED, HIGH); delay (2000);
  digitalWrite (RED,  LOW);
  digitalWrite (YELLOW, HIGH); delay (2000);
  digitalWrite (YELLOW,  LOW);
  digitalWrite (GREEN, HIGH); delay (2000);
  digitalWrite (GREEN,  LOW);

  pinMode (RED, INPUT);
  pinMode (YELLOW, INPUT);
  pinMode (GREEN, INPUT);

  return 0;
}